components/core/src/clp/streaming_compression/lzma/Decompressor.cpp

```
@@ -0,0 +1,362 @@
+    memset(m_decompression_stream, 0, sizeof(lzma_stream));
+}
+
+Decompressor::~Decompressor() {
+    delete m_decompression_stream;
+}
+
+void Decompressor::exact_read(char* buf, size_t num_bytes_to_read, size_t& num_bytes_read) {
+    auto errorcode = try_read(buf, num_bytes_to_read, num_bytes_read);
+    if (num_bytes_read != num_bytes_to_read) {
+        SPDLOG_ERROR("FAILED TO READ EXACTLY {} bytes", num_bytes_to_read);
+        throw;
```

cppcheck (2.10-2)

[error] Rethrowing current exception with 'throw;', it seems there is no current exception to rethrow. If there is no current exception this calls std

(rethrowNoCurrentException)